src/Migrate/deploymentscripts/Initialize-AzMigrateReplicationInfrastructure.ps1

```diff
@@ -0,0 +1,386 @@
+
+# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+The Initialize-AzMigrateReplicationInfrastructure deploys and configures the replication infrastructure used for server migration in the Azure Migrate project Resource Group.
+Requires: 
+Az.Resources
+Az.KeyVault
+Az.ServiceBus
+Az.Migrate
+
+Example Invocation:  .\Initialize-AzMigrateReplicationInfrastructure.ps1 -ResourceGroupName TestRG  -ProjectName TestProject -Vmwareagentless -TargetRegion centralus
+#>
+
+param(
+    [Parameter(Mandatory)]
+    [System.String]
+    # Specifies the Resource Group of the Azure Migrate Project in the current subscription.
+    ${ResourceGroupName},
+
+    [Parameter(Mandatory)]
+    [System.String]
+    # Specifies the name of the Azure Migrate project to be used for server migration.
+    ${ProjectName},
+
+    [Parameter(Mandatory)]
+    [System.Management.Automation.SwitchParameter]
+    # Specifies the server migration scenario for which the replication infrastructure needs to be initialized.
+    ${Vmwareagentless},
+
+    [Parameter(Mandatory)]
+    [System.String]
+    # Specifies the target Azure region for server migrations.
+    ${TargetRegion},
+
+    [Parameter()]
+    [System.String]
+    # Azure Subscription ID.
+    ${SubscriptionId}
+)
+
+process {
+        Import-Module -Name Az.Migrate
+        # Get/Set SubscriptionId
+        if(($SubscriptionId -eq $null) -or ($SubscriptionId -eq "")){
+            $context = Get-AzContext
+            $SubscriptionId = $context.Subscription.Id
+            if(($SubscriptionId -eq $null) -or ($SubscriptionId -eq "")){
+                throw "Please login to Azure to select a subscription."
+            }
+        }else{
+            Select-AzSubscription -SubscriptionId $SubscriptionId
+        }
+        $context = Get-AzContext
+        Write-Host "Using Subscription Id: ", $SubscriptionId
+        Write-Host "Selected Target Region: ", $TargetRegion
+        
+        $rg = Get-AzResourceGroup -Name $ResourceGroupName -ErrorVariable notPresent -ErrorAction SilentlyContinue
```

Comment:
seems this is target Resource group name not necessarily equal to Azure Migrate project RG. confused with the description provided above.